### PR TITLE
oauth2: fix Max-Age attribute of Set-Cookie response header

### PR DESCRIPTION
**Pull request by ggmoy**
Commit Message: oauth2: fix Max-Age attribute of Set-Cookie response header

Additional Description: The `Max-Age` attribute of `Set-Cookie` response header [indicates the number of seconds until the cookie expires](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie). Currently, we are assigning a value representing **Seconds Since the Epoch** to the `Max-Age` attribute. This is not correct and causes cookies to expire in ~53 years.

This PR updates the code to use the `expires_in` value received from the OAuth server response to set the `Max-Age` attribute.

Risk Level:
Testing: Adding unit tests for the fixed code and for the new runtime guard introduced by this PR
Docs Changes:
Release Notes:
Platform Specific Features:
Runtime guard: Adding `envoy.reloadable_features.oauth_use_standard_max_age_value` runtime guard
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by ggmoy** [comment]
Hey Folks, could someone tell me if I have to do something else here?

File: changelogs/current.yaml [hunk 1]
```diff
@@ -6,6 +6,11 @@ behavior_changes:
     OAuth filter now URL-encodes URL in query parameters. These query parameters are decoded, leaving intact character
     sequences that must remain encoded in URLs. This behavioral change can be temporarily reverted by setting runtime guard
     ``envoy.reloadable_features.oauth_use_url_encoding`` to false.
+- area: oauth2
```
Comment (yanavlasov) [inline changelogs/current.yaml:9]: nit: I think a note in the bug fixes section is enough.

Comment (ggmoy) [reply changelogs/current.yaml]: @yanavlasov, the entry has been removed from `behavior_changes`.

**Review by yanavlasov — DISMISSED** [review]

**Comment by yanavlasov** [comment]
Oops, please merge main and fix the merge conflict. I will ping other maintainers to review.

**Comment by yanavlasov** [comment]
@derekargueta PTAL as @snowp is OOO

**Comment by yanavlasov** [comment]
Ping @snowp or @derekargueta

**Review by snowp — APPROVED** [review]
LGTM, thanks for the fix!

**Comment by ggmoy** [comment]
> LGTM, thanks for the fix!

@snowp/@yanavlasov, do I need to do something to apply this to the release branch? How does this work?

**Pull request merged**